documentation/puppet-api/v3/environment_modules.markdown

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Environment Modules"
+canonical: "/puppetserver/latest/puppet-api/v3/environment_modules.html"
+---
+
+[deprecated WEBrick Puppet master]: https://docs.puppet.com/puppet/latest/reference/services_master_webrick.html
+[`auth.conf`]: ../../config_file_auth.markdown
+[`puppetserver.conf`]: ../../config_file_puppetserver.markdown
+
+The environment modules API will return information about what modules are
+installed for the requested environment.
+
+This endpoint is available only when the Puppet master is running Puppet Server, not
+on Ruby Puppet masters, such as the [deprecated WEBrick Puppet master][]. It also ignores
+the Ruby-based Puppet master's authorization methods and configuration. See the
+[Authorization section](#authorization) for more information.
+
+## `GET /puppet/v3/environment_modules?environment=:environment`
+
+Making a request with no query parameters is not supported and returns an HTTP 400 (Bad
+Request) response.
+
+### Supported HTTP Methods
+
+GET
+
+### Supported Formats
+
+JSON
+
+### Query Parameters
+
+Provide one parameter to the GET request:
+
+* `environment`: Only the modules information pertaining to the specified
+environment will be returned for the call.
```

Comment:
I'd rewrite this:

```
* `environment`: Request information about modules pertaining to the specified environment only.
```

Also, is this parameter mandatory? If so, we should note that (L34, "Provide one *mandatory* parameter ...).

If it's not required, is there a default value or response?